packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/io.ts

```diff
@@ -64,7 +65,21 @@ export class PnpQuotaIO extends IO<PnpQuotaRequest> {
   }
 
   async authenticate(request: Request<{}, {}, PnpQuotaRequest>, logger: Logger): Promise<boolean> {
-    return authenticateUser(request, this.kit, logger, this.config.shouldFailOpen)
+    const { success, failedOpen } = await authenticateUser(
+      request,
+      this.kit,
+      logger,
+      this.config.shouldFailOpen
+    )
+
+    if (failedOpen) {
```

Comment:
wondering if it would be cleaner (and reduce code duplication across all of the PNP io `authenticate` functions) to have `authenticateUser` take in an optional `warnings` array and push the `FAILURE_TO_GET_DEK` warning there? I guess then we wouldn't need to deal with success vs. failing open everywhere else?